include/fuse_rl/unicycle_2d/ignition.h

```diff
@@ -0,0 +1,139 @@
+
+#include <geometry_msgs/PoseWithCovarianceStamped.h>
+#include <robot_localization/SetPose.h>
+#include <ros/ros.h>
+
+
+namespace fuse_rl
+{
+
+namespace unicycle_2d
+{
+
+/**
+ * @brief A fuse_rl ignition sensor designed to be used in conjunction with the unicycle 2D motion model.
+ *
+ * This class publishes a transaction that contains a prior on each state subvariable used in the unicycle 2D motion
+ * model (x, y, yaw, x_vel, y_vel, yaw_vel, x_acc, and y_acc). When the sensor is first loaded, it publishes a single
+ * transaction with the configured initial state and covariance. Additionally, whenever a pose is received, either
+ * on the set_pose service or the topic, this ignition sensor resets the optimizer then publishes a new transaction
+ * with a prior at the specified pose. Priors on (x_vel, y_vel, yaw_vel, x_acc, and y_acc) continue to use the values
+ * configured on the parameter server.
+ *
+ * Parameters:
+ *  - ~device_id (uuid string, default: 00000000-0000-0000-0000-000000000000) The device/robot ID to publish
+ *  - ~device_name (string) Used to generate the device/robot ID if the device_id is not provided
+ *  - ~initial_sigma (vector of doubles) An 8-dimensional vector containing the standard deviations for the initial
+ *                                       state values. The covariance matrix is created placing the squared standard
+ *                                       deviations along the diagonal of an 8x8 matrix. Variable order is
+ *                                       (x, y, yaw, x_vel, y_vel, yaw_vel, x_acc, y_acc).
+ *  - ~initial_state (vector of doubles) An 8-dimensional vector containing the initial values for the state.
+ *                                       Variable order is (x, y, yaw, x_vel, y_vel, yaw_vel, x_acc, y_acc).
+ *  - ~queue_size (int, default: 10) The subscriber queue size for the pose messages
+ *  - ~reset_service (string, default: "reset") The name of the reset service to call before sending a transaction
+ *  - ~set_pose_service (string, default: "set_pose") The name of the set_pose service to advertise
+ *  - ~topic (string, default: "set_pose") The topic name for received PoseWithCovarianceStamped messages
+ */
+class Ignition : public fuse_core::AsyncSensorModel
+{
+public:
+  SMART_PTR_DEFINITIONS(Ignition);
+  using ParameterType = parameters::Unicycle2DIgnitionParams;
+
+  /**
+   * @brief Default constructor
+   *
+   * All plugins are required to have a constructor that accepts no arguments
+   */
+  Ignition();
+
+  /**
+   * @brief Destructor
+   */
+  ~Ignition() = default;
+
+  /**
+   * @brief Triggers the publication of a new prior transaction at the supplied pose
+   */
+  void subscriberCallback(const geometry_msgs::PoseWithCovarianceStamped::ConstPtr& msg);
+
+  /**
+   * @brief Triggers the publication of a new prior transaction at the supplied pose
+   */
+  bool serviceCallback(robot_localization::SetPose::Request& req, robot_localization::SetPose::Request&);
```

Comment:
I feel a little strange about using the robot_localization package here. Anyone have any better ideas?

Reply:
Message types in ROS don't care about the actual name/type of the message, so long as the contents are identical. We've taken advantage of this internally. With that in mind, I think I'd opt for the following:

1. Create the identical service type in `fuse_rl`, and switch this callback to it, but add a deprecation notice when it's called.
2. Create a new service type that also has a return code (why I neglected that in the first place, I can't say).
3. Add a callback for the new service, and make this one wrap to it.

But that doesn't need to be for this PR. Feel free to leave this as-is, and throw that task my way.

Reply:
Like so (bool success, string reason)? Or would you prefer enum-like error codes? Or different names? Do we have a standard naming convention for service call return values?